### PR TITLE
[dashboards] Fix, update dashboard owners not propagating to charts o…

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

- [X] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Regression when updating a dashboard the dashboard's charts owners are not updated.
When changing the dashboard owners we expect that all the charts that belong to the dashboard to be updated with the same dashboard owners (and preserve their own also)

Note: Will add tests to cover this regression

### TEST PLAN
- Login with `admin`
- Create 2 users `gamma` and `alpha`
- Update a dashboard setting it's new owners with `gamma` and `alpha`
- Check that all the charts that belong to the dashboard get it's owners updated with `gamma`, `alpha` and `admin`

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@suddjian @willbarrett

### CODE REVIEW

File: superset/dashboards/dao.py [hunk 1]
```diff
@@ -47,6 +47,15 @@ def validate_update_slug_uniqueness(dashboard_id: int, slug: Optional[str]) -> b
             return not db.session.query(dashboard_query.exists()).scalar()
         return True
 
+    @staticmethod
+    def update_charts_owners(model: Dashboard, commit: bool = True) -> Dashboard:
+        owners = [o for o in model.owners]
```
Comment (villebro) [inline superset/dashboards/dao.py:52]: I think it would be more lint-friendly to have `owner` in the list comprehension

**Review by villebro — APPROVED** [review]
LGTM, small non-blocking comment

File: superset/dashboards/dao.py [hunk 2]
```diff
@@ -47,6 +47,15 @@ def validate_update_slug_uniqueness(dashboard_id: int, slug: Optional[str]) -> b
             return not db.session.query(dashboard_query.exists()).scalar()
         return True
 
+    @staticmethod
+    def update_charts_owners(model: Dashboard, commit: bool = True) -> Dashboard:
+        owners = [owner for owner in model.owners]
+        for slc in model.slices:
+            slc.owners = list(set(owners) | set(slc.owners))
+        if commit:
```
Comment (suddjian) [inline superset/dashboards/dao.py:55]: Perhaps this commit logic should be a method decorator, for DRYness.

**Comment by john-bodley** [comment]
Sorry for the late message. I'm a tad perplexed why when a user updates a dashboard they need to be added as an owner of all the charts associated with the dashboard as they're not explicitly changing the underlying charts. 

cc: @graceguo-supercat

**Pull request merged**